.github/workflows/scheduled.yml

```diff
@@ -43,33 +48,63 @@ jobs:
 
       - name: Generate matrix
         id: matrix
+        working-directory: ${{ steps.set-up-homebrew.outputs.repository-path }}
         run: |
-          formula="$(find "$(brew --repo "${TAP}")/Formula" -type f | shuf -n "${TEST_COUNT}" | xargs -I{} basename {} .rb)"
-          # shellcheck disable=SC2086
+          # Index all formulae so that we test:
+          #   - formulae 1 to 50 on the first day of the year
+          #   - formulae 51 to 100 on the second day of the year
+          #   - formulae 101 to 150 on the third day of the year
+          #   - ...
+          # This works fine as long as we have fewer than 365 * TEST_COUNT formulae.
+          mapfile -t formulae < <(find Formula -type f -execdir basename -s '.rb' {} + | sort)
```

Comment:
Incidentally, BSD versions of these utilities are _slow_.
```
❯ time (/usr/bin/find Formula -type f -execdir /usr/bin/basename -s '.rb' {} + | /usr/bin/sort >/dev/null)
( /usr/bin/find Formula -type f -execdir /usr/bin/basename -s '.rb' {} + |  >)  2.56s user 4.29s system 82% cpu 8.303 total
❯ time (gfind Formula -type f -execdir gbasename -s '.rb' {} + | gsort >/dev/null)
( gfind Formula -type f -execdir gbasename -s '.rb' {} + | gsort > /dev/null;   0.03s user 0.03s system 87% cpu 0.063 total
```

Reply:
I think only `basename` is quite slow. May just use bash parameter expansion instead.

Reply:
Doesn't matter on the runner, since it's Linux, so it'll use GNU find/basename/sort.

Reply:
Interesting. May be GNU find has a more efficient piping through of data. If you run GNU tools through `|` or for loop, it is also slow. Most of time is recovered via parameter expansion.

Reply:
It's `-execdir` that's slow.
```
❯ time (/usr/bin/find Formula -type f -exec /usr/bin/basename -s '.rb' {} + | /usr/bin/sort >/dev/null)
( /usr/bin/find Formula -type f -exec /usr/bin/basename -s '.rb' {} + |  > ; )  0.01s user 0.01s system 102% cpu 0.023 total
```